src/Modules/Guilds/Hooks/useTimeDetail.ts

```diff
@@ -0,0 +1,51 @@
+import moment from 'moment';
+import { useTranslation } from 'react-i18next';
+import { ProposalState } from 'types/types.guilds.d';
+import {
+  getTimeDifferenceFromCurrentTimeHumanized,
+  isBeforeCurrentTime,
+} from 'utils/time/time';
+import { useGuildConfig } from './useGuildConfig';
+
+const useTimeDetail = (
+  guildId: string,
+  status: ProposalState,
+  endTime: moment.Moment
+) => {
+  const { t } = useTranslation();
+  const { data: guildConfig } = useGuildConfig(guildId);
+
+  let endTimeDetail = '';
+  let endTimeMoment: moment.Moment;
+
+  if (!endTime) return {};

```

Comment:
We could move the return before the variable initialization, or initializing both variables as `null` and returning `{ endTimeMoment, endTimeDetail }`

That way, we would have `null` instead of `undefined`, that I think looks better.

What do you think?